src/c_lib/base/base_model.pyx

```diff
@@ -206,7 +206,7 @@ def core_simulator(method,
 
     # -------------------------------------------------------------------------
     # sample __________________________________________________________________
-    for spin_sys in spin_systems:
+    for spin_sys, segments, weights in zip(spin_systems, transition_pathways, transition_weights):

```

Comment:
Have you tested the performance impact when using zip?

Reply:
It looks like a `zip` iterator creases more Python objects than a simple iteration over an index. Index iteration slightly faster taking 2.83 ms vs 2.97 ms per simulation for a large coupled spin system.

Will switch over to index iteration